### ISSUE TITLE
Escape user-supplied content in narrative HTML output

### ISSUE THREAD

**Issue by tuvelofstrom** [Moffran/calibrated_explanations#222]
## Problem

`to_narrative(output_format="html")` interpolates keys and values into the HTML table
without escaping. `NarrativePlotPlugin._format_as_html`
(`src/calibrated_explanations/viz/narrative_plugin.py:463-505`) builds the table with
f-strings: header cells at line 486 (`{key}`) and body cells at line 500 (`{value}`).
Values only have `"\n"` replaced by `<br>`; no `html.escape` is applied anywhere in the
module.

Feature names, class labels and categorical values are user-supplied, so any markup
they contain is emitted as live HTML (e.g. when displayed in a notebook).

## Reproduction

```python
import numpy as np
from sklearn.ensemble import RandomForestClassifier
from calibrated_explanations import WrapCalibratedExplainer

rng = np.random.default_rng(0); X = rng.random((120, 2)); y = (X[:, 0] > .5).astype(int)
w = WrapCalibratedExplainer(RandomForestClassifier(n_estimators=10, random_state=0))
w.fit(X[:60], y[:60])
w.calibrate(X[60:100], y[60:100], feature_names=["<img src=x onerror=alert(1)>", "b&c"])
out = w.explain_factual(X[100:101]).to_narrative(output_format="html")
print("<img src=x" in out, "&lt;img" in out)   # True False
```

## Acceptance criteria

* [ ] Header and body cell content is escaped with `html.escape` before insertion.
* [ ] The intentional `\n` -> `<br>` conversion is applied *after* escaping.
* [ ] Tests (`test_should_escape_markup_when_feature_label_contains_html`) cover
  feature names, class labels and categorical values containing `<`, `>`, `&`, quotes.
* [ ] Non-HTML output formats (`text`, `markdown`, `dict`, `dataframe`) are unchanged.
* [ ] CHANGELOG entry under the patch release.

## Related

Identified during review of the narrative provider design (`narrative-plugin` branch,
work item N12). Fixing it does not depend on that design.
